src/MIDebugEngine/AD7.Impl/AD7Property.cs

```diff
@@ -141,10 +142,32 @@ public int EnumChildren(enum_DEBUGPROP_INFO_FLAGS dwFields, uint dwRadix, ref Gu
                 {
                     _engine.DebuggedProcess.Natvis.WaitDialog.ShowWaitDialog(_variableInformation.Name);
                     var children = _engine.DebuggedProcess.Natvis.Expand(_variableInformation);
-                    DEBUG_PROPERTY_INFO[] properties = new DEBUG_PROPERTY_INFO[children.Length];
-                    for (int i = 0; i < children.Length; i++)
+
+                    // Count number of children that fit filter (results saved in "fitsFilter")
+                    int propertyCount = children.Length;
+                    bool[] fitsFilter = null;
+                    if (!string.IsNullOrEmpty(pszNameFilter))
+                    {
```

Comment:
Rather than doing it this way, I wonder if we want OpenDebugAD7 keep a collection of children by name. This is what vsdbg does (see m_variablesChildren in the Concord repo).

Reply:
Should we remove these changes now? I think it's fine to add, but also this is no longer used.

Reply:
Up to you.

Reply:
I'll leave them